core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -563,9 +563,15 @@ private Expression getConvertExpression(
     case TINYINT:
     case SMALLINT: {
       if (SqlTypeName.NUMERIC_TYPES.contains(sourceType.getSqlTypeName())) {
+        Type javaClass = typeFactory.getJavaClass(targetType);
```

Comment:
This change is only indirectly related to this PR: without this change, the code generated for ROW casts crashes at runtime.

Reply:
Why does it crash? It feels a bit strange